src/stan_math_backend/Lower_functions.ml

```diff
@@ -162,39 +161,25 @@ let mk_extra_args templates args =
 let lower_args extra_templates extra args variadic =
   let args, variadic_args =
     match variadic with
-    | `ReduceSum -> List.split_n args 3
-    | `VariadicHOF x -> List.split_n args x
-    | `None -> (args, []) in
+    | ReduceSum -> List.split_n args 3
+    | VariadicHOF x -> List.split_n args x
+    | FixedArgs -> (args, []) in
   let arg_strs =
     args
     @ mk_extra_args extra_templates extra
     @ [(Pointer (TypeLiteral "std::ostream"), "pstream__")]
     @ variadic_args in
   arg_strs
 
-let add_functor_decl functors (name : string)
-    (param : template_parameter option) (f : fun_defn) =
-  let f existing =
-    match existing with
-    | None -> make_struct_defn ~param ~name ~body:[FunDef f] ()
-    | Some sd -> {sd with body= sd.body @ [FunDef f]} in
-  Hashtbl.update functors name ~f
-
```

Comment:
Actually it's the removal of this function that I think broke the overloading tests. We probably need something like it still (I think any version which doesn't have this sort of global variable ends up being O(N^2) because you need to search for overloads for each struct you generate)